### DISCUSSION TITLE
[Proposal] Add a helper method to get the Unix file permissions off of ZipEntry

### DISCUSSION THREAD

**Discussion by IvanLH** [icsharpcode/SharpZipLib#692] [Q&A]
### Steps to reproduce
1. Zip a file on Linux using Info-Zip
2. Unzip using the library
3. There's no easy or well documented way to retrieve the file permissions whose structure is this: https://unix.stackexchange.com/questions/14705/the-zip-formats-external-file-attribute

### Expected behavior
I am proposing to add a method called int GetUnixStyleFilePermissions() to ZipEntry which will get the necessary bits and return them as int.

### Actual behavior
No such functionality exists

### Version of SharpZipLib
1.2.0
### Obtained from (only keep the relevant lines)
- Package installed using NuGet

**Comment by piksel** [comment]
`ZipEntry.ExternalFileAttributes` is what you are looking for.

**Reply by IvanLH** [reply]
I may be reading the code wrong, but it seems the ExternalFileAttributes property is returning the full 32 bits of the external attributes as an Int. And according to what I've read, only 9 of those are permission bits on Unix files. The end user of the library can still get those, but it's not documented in the library and I think a helper method to do it for them may be warranted. 

Code from ZipEntry

```
/// <summary>
/// Get/Set external file attributes as an integer.
/// The values of this are operating system dependent see
/// <see cref="HostSystem">HostSystem</see> for details
/// </summary>
public int ExternalFileAttributes
{
	get => (known & Known.ExternalAttributes) == 0 ? -1 : externalFileAttributes;

	set
	{
		externalFileAttributes = value;
		known |= Known.ExternalAttributes;
	}
}
```

**Reply by piksel** [reply]
Yeah, it's the full 4 byte value, since the specification doesn't declare how they are used (except for when `version made by` indicates DOS). Examples of how to read them could be added to the wiki though and would be a welcomed contribution!

**Reply by IvanLH** [reply]
I can add that, but I am also open to adding a helper method, since this use case seems common enough :)

**Reply by piksel** [reply]
The reason I'm reluctant about it is that my main guide line as maintainer is to be compliant with the official ZIP specification: https://p1k.se/appnote.md

Perhaps you could implement it as an extension method and put it in the namespace `ICSharpCode.SharpZipLib.Zip.Unix`?

**Reply by emmayspark** [reply]
To follow up with this, I tried to get a file permission but zipEntry.externalFileAttributes returned -1. I'm not sure this is expected - I zipped the files using Info-zip on Linux and unzipped it using the SharpZipLib.
